### PR TITLE
Start ! to allow running of OS commands

### PR DESCRIPTION
**Pull request by rocky**

### CODE REVIEW

File: mathicsscript/__main__.py [hunk 1]
```diff
@@ -333,6 +333,12 @@ def main(
 
             evaluation = Evaluation(shell.definitions, output=TerminalOutput(shell))
             query, source_code = evaluation.parse_feeder_returning_code(shell)
+            if len(source_code) and source_code[0] == "!":
+                os.system(source_code[1:])
```
Comment (GarkGarcia) [inline mathicsscript/__main__.py]: We should probably place an option somewhere to disable this functionality, shouldn't we? For example, someone running mathics on a webserver wouldn't want to allow strangers to run arbitrary scripts on their machine.

**Review by GarkGarcia — APPROVED** [review]
Seems good to me, but perhaps we should consider implementing this as a builtin in core (?)

File: mathicsscript/__main__.py [hunk 2]
```diff
@@ -333,11 +335,6 @@ def main(
 
             evaluation = Evaluation(shell.definitions, output=TerminalOutput(shell))
             query, source_code = evaluation.parse_feeder_returning_code(shell)
-            if len(source_code) and source_code[0] == "!":
-                os.system(source_code[1:])
-                # Should we test exit code for adding to history?
```
Comment (GarkGarcia) [inline mathicsscript/__main__.py]: We could return `$Failure` or atleast print an error if the exit code isn't zero.

**Review by GarkGarcia — APPROVED** [review]

**Comment by rocky** [comment]
> Seems good to me, but perhaps we should consider implementing this as a builtin in core (?)

It is already in core in as much as it can be See https://github.com/mathics/Mathics/pull/1085

This has to handled by the front-end since it needs to act like it is detected before scanning and parsing occurs.

**Pull request merged**